[master] Fix cmdmod _run log level checking bug

jopelima opened the pull request:
Early in the _run() function, on the command execution module, the [_check_loglevel()](https://github.com/saltstack/salt/blob/master/salt/modules/cmdmod.py#L335) function is called, and the value of the output_loglevel variable is rewritten with the output of this function.

The [_check_loglevel()](https://github.com/saltstack/salt/blob/master/salt/modules/cmdmod.py#L190-L191) function seems to sanitise the input variable, but if the requested log level is "quiet", it returns None.

Later on, in the _run() function, there are checks comparing [output_loglevel != "quiet"](https://github.com/saltstack/salt/blob/master/salt/modules/cmdmod.py#L769), which is always true at that point, since the _check_loglevel() function set the output_loglevel to None if the input parameter was "quiet".

This leads to error messages being logged if the output of the executed command contains non-decodable data, using the provided encoding (e.g. https://github.com/saltstack/salt/blob/4b652eb666045ccff231c49065f47f74b91ee4ab/salt/modules/cmdmod.py#L788 ) even though this call requested no logs to be produced.

This commit fixes the erroneous output_loglevel checks, to compare against None rather than "quiet" after the output_loglevel variable has been processed.

jopelima committed:
Fix cmdmod _run log level checking bug

Early in the _run() function, on the command execution module, the _check_loglevel() function is called, and the value of the output_loglevel variable is rewritten with the output of this function.

The _check_loglevel() function seems to sanitise the input variable, but if the requested log level is "quiet", it returns None.

Later on, in the _run() function, there are checks comparing `output_loglevel != "quiet"`, which is always true at that point, since the _check_loglevel() function set the output_loglevel to None if the input parameter was "quiet".

This leads to error messages being logged (e.g. https://github.com/saltstack/salt/blob/4b652eb666045ccff231c49065f47f74b91ee4ab/salt/modules/cmdmod.py#L788), evven though this module was requested to produce no logs.

This commit fixes the erroneous output_loglevel checks, to compare against None rather than "quiet" after the output_loglevel variable has been processed.

twangboy reviewed (changes requested):
PRs require a changelog and tests prior to being merged.

Changelog:
https://docs.saltproject.io/en/master/topics/development/changelog.html

Tests:
https://docs.saltproject.io/en/master/topics/development/tests/index.html

This also needs to be rebased and conflicts resolved

dwoz committed:
Add changelog and test for cmdmod _run quiet loglevel fix

Add changelog/65904.fixed.md and a unit test
(test_run_all_binary_replace_quiet) that reproduces the pre-fix behaviour:
non-decodable stdout/stderr with output_loglevel="quiet" must not emit
log.error messages.
